astropy/table/table.py

```diff
@@ -2447,7 +2447,9 @@ def argsort(self, keys=None, kind=None):
 
         # use index sorted order if possible
         if keys is not None:
-            index = get_index(self, self[keys])
+            # Creating a sliced copy of table
+            table_cp = self.__class__([self[key] for key in keys], copy=False)
+            index = get_index(self, table_cp)
```

Comment:
Let's take the opportunity here to actually fix the underlying problem that `get_index` is written in a non-ideal way.  The only reason it needs the table copy is to get the column names, so why not just provide the names! This is an example of a back-compatible API change for improved performance.

I also don't like the confusing use of `col` and `column` in the original code.  In table code in general, `col` should be a column object, while `name` is the name of a column.  A frequent idiom for consistency and code understanding is using a plural like `names` for a list or array of things and then iterating over the singular like `name`.
```
def get_index(table, table_copy=None, names=None):
    """
    Inputs a table and some subset of its columns, and
    returns an index corresponding to this subset or None
    if no such index exists.

    TODO: fix the docstring description above to be more accurate.

    Parameters
    ----------
    table : `Table`
        Input table
    table_copy : `Table`, optional
        Subset of the columns in the ``table`` argument
    names : list, tuple, optional
        Subset of column names in the ``table`` argument

    Returns
    -------
    TODO

    """
    # TODO: validate that one and only one of table_copy and keys is provided

    if names is not None:
        names = set(names)
    else:
        names = set(table_copy.colnames)

    if not names <= set(table.colnames):
        raise ValueError('{} is not a subset of table columns'.format(names))

    for name in names:
        for index in table[name].info.indices:
            if set([col.info.name for col in index.columns]) == names:
                return index

    return None
```